app/models/policy_version.rb

```diff
@@ -86,6 +86,58 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    log_versions_off_limit
+    delete_versions_off_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
```

Comment:
@shulifink

Reply:
@eranha  - can you show an example of the error with all the variables filled in?